impl/src/main/java/org/jboss/weld/bean/builtin/InstanceImpl.java

```diff
@@ -101,9 +101,17 @@ public T get() {
         if (bean != null) {
             return getBeanInstance(bean);
         } else if (isUnsatisfied()) {
-            throw BeanManagerLogger.LOG.unresolvableElement("Type: " + getType() + "; Qualifiers: " + getQualifiers());
+            throw BeanManagerLogger.LOG.injectionPointHasUnsatisfiedDependencies(ip,
+                    Formats.formatAnnotations(ip.getQualifiers()),
+                    Formats.formatInjectionPointType(ip.getType()),
+                    Formats.formatAsStackTraceElement(ip),
+                    InjectionPoints.getUnsatisfiedDependenciesAdditionalInfo(ip, getBeanManager()));
         } else {
-            throw BeanManagerLogger.LOG.ambiguousBeansForDependency(WeldCollections.toMultiRowString(allBeans));
+            throw BeanManagerLogger.LOG.injectionPointHasAmbiguousDependencies(ip,
+                    Formats.formatAnnotations(ip.getQualifiers()),
+                    Formats.formatInjectionPointType(ip.getType()),
+                    Formats.formatAsStackTraceElement(ip),
+                    InjectionPoints.getUnsatisfiedDependenciesAdditionalInfo(ip, getBeanManager()));
```

Comment:
This is wrong. Ambiguous dependency means more than one beans. The original `WeldCollections.toMultiRowString(allBeans)` was correct.